src/tqec/generation/circuit.py

```diff
@@ -54,6 +48,7 @@ def generate_circuit(
     # Index 0 is "no plaquette" by convention.
     _indices = list(range(len(plaquettes) + 1))
     template_plaquettes = template.instanciate(*_indices)
+    plaquette_increments = template.get_template_increments(*_indices)
```

Comment:
This ends up calling the same method twice under-the-hood. Maybe make the `build_array` method public (renaming it with a more descriptive name) and use it here?

Reply:
I realized this, too. My solution would have been to cache this, because I didn't want to change how the `instanciate` method works. I'll make it public but keep the others.

Reply:
Maybe make the `_build_array` method "public" again by renaming it to `build_array` (or a more descriptive name) and:
```suggestion
    template_plaquettes, plaquette_increments = template.build_array(*_indices)
```
?